### PR TITLE
Update identity settings of logic app standard to support user identity

### PR DESCRIPTION
**Pull request by tjezek**

### CODE REVIEW

File: modules/azure/logic_app_standard/variables.tf [hunk 1]
```diff
@@ -34,10 +34,13 @@ variable "enabled" {
...
 }
 
-variable "use_managed_identity" {
-  type        = bool
-  description = "Use Managed Identity for this logic app"
-  default     = false
+variable "identity" {
+  type = object({
+    identity_type = string, # use one of these values: "SystemAssigned", "UserAssigned", "SystemAssigned, UserAssigned"
+    identity_ids  = optional(list(string), [])
```
Comment (tom-reinders) [inline modules/azure/logic_app_standard/variables.tf:40]: Have you checked is the default of `[]` is the same as not adding an identity_ids at all?

**Review by tom-reinders — CHANGES_REQUESTED** [review]

File: modules/azure/logic_app_standard/main.tf [hunk 2]
```diff
@@ -31,9 +31,10 @@ resource "azurerm_logic_app_standard" "app" {
   version             = var.logic_app_version
 
   dynamic "identity" {
-    for_each = var.use_managed_identity ? [1] : []
+    for_each = var.identity != null ? [1] : []
```
Comment (tom-reinders) [inline modules/azure/logic_app_standard/main.tf:34]:
```suggestion
    for_each = var.use_managed_identity ? [1] : []
```

File: modules/azure/logic_app_standard/main.tf [hunk 3]
```diff
@@ -31,9 +31,10 @@ resource "azurerm_logic_app_standard" "app" {
...
     content {
-      type = "SystemAssigned"
+      type         = var.identity.identity_type
+      identity_ids = var.identity.identity_ids
```
Comment (tom-reinders) [inline modules/azure/logic_app_standard/main.tf:37]:
```suggestion
      type         = var.identity_type
      identity_ids = var.identity_ids
```

**Review by tom-reinders — CHANGES_REQUESTED** [review]

File: modules/azure/logic_app_standard/variables.tf (hunk 1 above)
Comment (tjezek) [reply modules/azure/logic_app_standard/variables.tf:40]: Yes, terraform provider sets empty array inside state if identity_ids is null.

File: modules/azure/logic_app_standard/variables.tf [hunk 4]
```diff
@@ -34,10 +34,13 @@ variable "enabled" {
...
+  })
+  default     = null
+  description = "Logic App Identity settings"
```
[the remark this answers is not in the document]
Comment (tjezek) [reply modules/azure/logic_app_standard/variables.tf]: Ok, I updated PR (with commit https://github.com/recognizegroup/terraform/pull/317/commits/b6d06c84b5d625dbc5ff5d38420b65c0b6e572db).

**Review by tom-reinders — APPROVED** [review]

**Pull request merged**